### PR TITLE
Add IP version selection to shortcut advanced options

### PR DESCRIPTION
**Pull request by mdzeldemt**
Hi! First of all, thank you for this app - I use it a lot. Also, the code is really clean, so it was easy to get through it quickly. I want to make a tiny feature suggestion, if I may.

### Changes in this PR
This PR adds an option for selecting the IP version (IPv4 or IPv6) used by a shortcut when resolving hostnames (basically the equivalent to adding either option `-4` or `-6` from [curl](https://curl.se/docs/manpage.html)). If no option is selected (which is the default), the shortcut will behave as it does currently.

#### What is updated
- The _Protocol Version_ setting (with options _determine automatically_, _IPv4_ and _IPv6_) is added to the _Advanced Technical Settings_ screen
- The choice of protocol version takes effect when invoking the shortcut (if 'determine automatically' is selected, the behavior is the same as in the current production version)
- The `-4` and `-6` options can now be parsed from a curl command when using the "import from cURL command" option
- The Protocol Version option will now be present when exporting the shortcut to either curl or JSON (in the latter case, it will show up as `protocolVersion`)

### More context (why I'm making this suggestion)
I'm used to resort to sites such as [ifconfig.me](https://ifconfig.me/) to query my global IP address -- it's really easy to type `curl ifconfig.me` on a terminal. The version of the IP address returned by these sites (or at least the ones I'm familiar with) is the same as the one used for the hostname resolution, which on most systems defaults to IPv6. However,  more often than not, I need the IPv4, so I actually need to invoke  `curl -4 ifconfig.me`. It would be really handy to be able to do this with the app, but currently this option is not available.

Admittedly, this is a pretty small use case, though maybe there are others that I'm not foreseeing right now.

### What I tested
- Create a shortcut with Protocol Version set to _determine automatically_/_IPv4_/_IPv6_ and verify that the response matches this choice
- Import a shortcut from a curl command with either the `-4` or `-6` flags (or both)
- Export a shortcut for each of the possible choices to both curl or JSON

This is my first PR here, so I will leave a couple of comments with a few clarifications.

### CODE REVIEW

File: HTTPShortcuts/curl_command/src/main/kotlin/ch/rmy/curlcommand/CurlCommand.kt [hunk 1]
```diff
@@ -122,6 +126,16 @@ class CurlCommand internal constructor() : Serializable {
...
 
+        fun ipVersion4() = also {
+            curlCommand.ipVersion4 = true
+            curlCommand.ipVersion6 = false
+        }
+
+        fun ipVersion6() = also {
+            curlCommand.ipVersion6 = true
+            curlCommand.ipVersion4 = false
+        }
```
Comment (mdzeldemt) [inline HTTPShortcuts/curl_command/src/main/kotlin/ch/rmy/curlcommand/CurlCommand.kt:137]: I'm following the convention stated at the [curl manpage](https://curl.se/docs/manpage.html):
> Providing --ipv6 multiple times has no extra effect.
> This option is mutually exclusive with -4, --ipv4

You can invoke `curl` with both the `-4` and `-6` flags. If done so, only the last one will be considered. This is also true of this implementation.

File: HTTPShortcuts/app/src/main/res/values-es/strings.xml [hunk 2]
```diff
@@ -257,6 +257,8 @@ Haga clic en el botón + para añadir una variable.</string>
     <string name="import_in_progress">Importando…</string>
     <!-- Label, for option to pick the timeout of a shortcut. clicking it opens a picker dialog -->
     <string name="label_timeout">Se acabó el tiempo</string>
+    <!-- Label, for dropdown picker, allowing to select the Internet Protocol version used for hostname resolution by a shortcut -->
+    <string name="label_ip_version">Versión del protocolo</string>
```
Comment (mdzeldemt) [inline HTTPShortcuts/app/src/main/res/values-es/strings.xml:261]: I only speak English and Spanish, so I can only contribute with text in these languages

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/http/HttpClientFactory.kt [hunk 3]
```diff
@@ -97,6 +102,14 @@ constructor() {
                 throw InvalidProxyException(e.message!!)
             }
         }
+        .runIfNotNull(ipVersion) {
+            dns { hostname ->
+                when (it) {
+                    IpVersion.V4 -> Dns.SYSTEM.lookup(hostname).filterIsInstance<Inet4Address>()
+                    IpVersion.V6 -> Dns.SYSTEM.lookup(hostname).filterIsInstance<Inet6Address>()
+                }
+            }
```
Comment (mdzeldemt) [inline HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/http/HttpClientFactory.kt:111]: This is the most relevant change

**Review by Waboodoo — APPROVED** [review]
Thanks a lot for your contribution, this looks great. I'll include this in the next release

**Pull request merged**